node/node.go

```diff
@@ -433,6 +430,14 @@ func makeNode(
 		},
 	}
 
+	if cfg.Mode == config.ModeValidator {
+		pubKey, err := privValidator.GetPubKey(ctx)
+		if pubKey == nil || err != nil {
+			return nil, fmt.Errorf("can't get pubkey: %w", err)
+		}
+		node.rpcEnv.PubKey = pubKey
+	}
+
 	node.rpcEnv.P2PTransport = node
```

Comment:
This is some weird shit I remember coming across that relates to querying `NodeInfo` but something we can do in a follow up issue perhaps

Reply:
yeah, definitely.